src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Discovery.cs

```diff
@@ -122,7 +122,7 @@ public ValueTask HandleAsync(HandleConfigurationResponseContext context)
                     context.Configuration.GrantTypesSupported.Add(GrantTypes.RefreshToken);
                 }
 
-                else if (context.Registration.ProviderType is ProviderTypes.Auth0 or ProviderTypes.Microsoft)
+                else if (context.Registration.ProviderType is ProviderTypes.Auth0 or ProviderTypes.Huawei or ProviderTypes.Microsoft)
```

Comment:
Are we sure Huawei supports the implicit flow?

https://developer.huawei.com/consumer/cn/doc/HMSCore-Guides/open-platform-oauth-0000001053629189 doesn't seem to list it and the `response_types_supported` node of the discovery document only lists `code` as a supported value.

Reply:
Sorry that the current code state is draft. They do not support implicit flow. I will update the code after everything I am not sure is confirmed.